publisher-sdk/src/main/java/com/criteo/publisher/csm/MetricRequest.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.csm
+
+import com.criteo.publisher.annotation.OpenForTesting
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@OpenForTesting
+@JsonClass(generateAdapter = true)
+data class MetricRequest internal constructor(
+    val feedbacks: List<MetricRequestFeedback>,
+    @Json(name = "wrapper_version")
+    val wrapperVersion: String,
+    @Json(name = "profile_id")
+    val profileId: Int
+) {
+  constructor(
+      metrics: Collection<Metric>,
+      sdkVersion: String,
+      profileId: Int
+  ) : this(metrics.map { MetricRequestFeedback(it) }, sdkVersion, profileId)
+
+  @OpenForTesting
+  @JsonClass(generateAdapter = true)
+  data class MetricRequestSlot(
+      val impressionId: String?,
```

Comment:
Finally, with kotlin null-check, I'm wondering if this field should be non-nullable. Could you try removing the nullability and check if there is any compilation/test errors ?

Reply:
Sure. Just tested, no compilation and no unit test errors